### ISSUE TITLE
how can  i ingest logs to azure 21v cloud or gov cloud?

### ISSUE THREAD

**Issue by ql3xHd630** [fluent/fluent-bit#9381]
how can  i ingest logs to azure 21v cloud or gov cloud?
i can not find some field to set azure cloud for log inegstion api 
![image](https://github.com/user-attachments/assets/01870aa6-2908-454c-9157-cbf4dd7df894)

**Comment by ql3xHd630** [comment]
![image](https://github.com/user-attachments/assets/80d147bf-46ba-4f8f-b10f-d655e566d553)
the default cloud app authentication endpoint is Azure global cloud

**Comment by incidrthreat** [comment]
@ql3xHd630 Not sure if this is still an issue on your end or not, but I came across your issue and was able to get it working with a build from source.  This might help others who come looking for answers.  I plan on either raising an issue with the team or submitting my own PR/MR with some edits to make it an optional input with the `.com` urls being the default.  We attempted both plugins and ultimately went with the `azure` plugin for our env to send logs directly to Azure Log Analytics Workspace/Sentinel. 

Steps I took:
- Cloned the fluent-bit repo
- Edited `plugins/out_azure/azure.h`
  - Changed `FLB_AZURE_HOST` to `".ods.opinsights.azure.us"`

OR
 
- Edited `plugins/out_azure_logs_ingestion/azure_logs_ingestion.h`
  - Changed `FLB_AZ_LI_AUTH_SCOPE` to `"https://monitor.azure.us/.default"` and `FLB_AZ_LI_AUTH_URL_TMPLT` to `"https://login.microsoftonline.us/"`
- `docker build -t <your tag:version here> --target=production -f dockerfiles/Dockerfile .`


Hope this helps!
